NitroxClient/MonoBehaviours/Gui/Chat/PlayerChat.cs

```diff
@@ -0,0 +1,156 @@
+            {
+                transform.GetChild(0).GetComponent<Image>(), transform.GetChild(1).GetComponent<Image>(), transform.GetChild(3).GetComponent<Image>()
+            };
+            yield return new WaitForEndOfFrame();
+        }
+
+        public void WriteLogEntry(ChatLogEntry chatLogEntry)
+        {
+            chatLogEntry.MessageText = SanitizeMessage(chatLogEntry.MessageText);
+
+            if (entries.Count == MESSAGE_LIMIT)
+            {
+                Destroy(entries.Values.First());
+                entries.Remove(entries.Keys.First());
+            }
+
+            GameObject newChatLogEntry = Instantiate(logEntryPrefab, logEntryPrefab.transform.parent);
+            newChatLogEntry.GetComponent<PlayerChatLogItem>().ApplyOnPrefab(chatLogEntry);
+            entries.Add(chatLogEntry, newChatLogEntry);
+
+            StartCoroutine(UpdateChatEntrySpacing());
+        }
+
+        /// Updates the layout sorting algorithm from Unity to prevent "loss" of text messages.
+        private IEnumerator UpdateChatEntrySpacing()
+        {
+            yield return new WaitForEndOfFrame();
+            foreach (HorizontalOrVerticalLayoutGroup layoutGroup in layoutGroups)
+            {
+                layoutGroup.enabled = false;
+            }
+            yield return new WaitForEndOfFrame();
+            foreach (HorizontalOrVerticalLayoutGroup layoutGroup in layoutGroups)
+            {
+                layoutGroup.enabled = true;
+            }
+        }
+
+        public bool ValidateChatMessage(out string text)
+        {
+            text = inputField.text;
+            return inputField.text.Trim() != "";
+        }
+
+        public void ResetInputField() => inputField.text = "";
+
+        public void Show()
+        {
+            PlayerChatInputField.ResetTimer();
+            StartCoroutine(ToggleTransparencyCanvasGroup(true));
+        }
+        public void Hide()
+        {
+            StartCoroutine(ToggleTransparencyCanvasGroup(false));
+        }
+
+        public void Select()
+        {
+            base.Select(true);
+            inputField.Select();
+            inputField.ActivateInputField();
+        }
+
+        public void Deselect()
+        {
+            base.Deselect();
+            EventSystem.current.SetSelectedGameObject(null);
+        }
+
+        private static string SanitizeMessage(string message)
+        {
+            message = message.Trim();
+            return message.Length < LINE_CHAR_LIMIT ? message : message.Substring(0, LINE_CHAR_LIMIT);
+        }
+
+        private void ToggleTransparency()
+        {
+            float alpha = transparent ? 1f : TOGGLED_TRANSPARENCY;
+            transparent = !transparent;
+
+            foreach (Image backgroundImage in backgroundImages)
+            {
+                backgroundImage.CrossFadeAlpha(alpha, 0.5f, false);
+            }
+        }
+
+        private IEnumerator ToggleTransparencyCanvasGroup(bool fadeIn)
```

Comment:
Rename to indicate it will fade the whole chat. Not just change transparency (it will hide or show all the way). Also `canvasGroup` isn't clear enough imo.

Example:
```suggestion
        private IEnumerator ToggleChatFade(bool fadeIn)
```

Reply:
The problem is that the Unity component is called `CanvasGroup`, so don't know how to name it. better without adding useless info